pkg/dwarf/line/state_machine.go

```diff
@@ -231,14 +231,15 @@ func (lineInfo *DebugLineInfo) stateMachineFor(basePC, pc uint64) *StateMachine
 		sm = newStateMachine(lineInfo, lineInfo.Instructions, lineInfo.ptrSize)
 	} else {
 		// Try to use the last state machine that we used for this function, if
-		// there isn't one or it's already past pc try to clone the cached state
+		// there isn't one, or it's already past pc try to clone the cached state
 		// machine stopped at the entry point of the function.
 		// As a last resort start from the start of the debug_line section.
 		sm = lineInfo.lastMachineCache[basePC]
 		if sm == nil || sm.lastAddress >= pc {
 			sm = lineInfo.stateMachineForEntry(basePC)
 			lineInfo.lastMachineCache[basePC] = sm
 		}
+		sm = sm.copy()
```

Comment:
This is wrong, the intent is to reuse the last machine and make it advance so that when we call PCToLine many times to get the line numbers for a disassembly it isn't quadratic.

Reply:
Sorry, I didn't notice it yesterday.

Reply:
Ack, yeah I forgot to revert this change. I was wondering for a moment if this cache was causing issues but it was a red herring. I’ll revert tomorrow.

Reply:
Done.